src/main/java/com/nirima/jenkins/plugins/docker/utils/PortUtils.java

```diff
@@ -174,7 +174,7 @@ public boolean execute() throws InterruptedException {
         private boolean executeOnce(final int thisTryNumber, final int totalTriesIntended) {
             final Connection sshConnection = new Connection(parent.host, parent.port);
             try {
-                sshConnection.connect(null, 0, sshTimeoutMillis, sshTimeoutMillis);
+                sshConnection.connect(null, sshTimeoutMillis, 0, sshTimeoutMillis);
```

Comment:
While I agree that passing a non-zero value as the `connectTimeout` argument is a good idea, I don't believe that passing `0` as the `readTimeout` argument is a good idea.
I think that passing zero anywhere will _either_ result in "no timeout" or "a timeout of zero" and both outcomes are bad (one results in non-robust code, the other results in the situation you reported & are trying to fix).

Reply:
there is a bug related to readTimeout JENKINS-48618

Reply:
It's not clear from that bug description where the bug is or what the proposed fix is.
The only circumstances where the docker-plugin _should_ be passing in a zero readTimeout is if the trilead library has a bug requiring it to be zero ... and if there is a bug then that should be fixed in the trilead code rather than modifying all callers to pass in a zero where a non-zero is preferred.

Reply:
sure, it should be fixed in the trilead-ssh2, put a 0 is a workaround